src/components/JobsPanelDataInputs/JobsPanelDataInputs.js

```diff
@@ -56,61 +62,88 @@ const JobsPanelDataInputs = ({
       }
     }
   }, [
-    pathScheme,
     inputsState.newInput.path.project.length,
     inputsState.projects.length,
     match.params.projectName,
-    projectStore.projects
+    projectStore.projects,
+    inputsState.newInput.path.pathType,
+    inputsState.selectedDataInput.data.path.value,
+    inputsState.selectedDataInput
   ])
 
   useEffect(() => {
     if (
       inputsState.artifacts.length === 0 &&
-      inputsState.newInputProjectPathEntered
+      inputsState.inputProjectPathEntered
     ) {
-      fetchArtifacts({ project: inputsState.newInput.path.project }).then(
-        artifacts => {
-          const artifactsList = artifacts
-            .map(artifact => ({
-              label: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? '',
-              id: artifact.link_iteration
-                ? artifact.link_iteration.db_key
-                : artifact.key ?? ''
-            }))
-            .filter(artifact => artifact.label !== '')
-
-          inputsDispatch({
-            type: inputsActions.SET_ARTIFACTS,
-            payload: artifactsList
-          })
-        }
-      )
+      fetchArtifacts({
+        project:
+          inputsState.newInput.path.project ||
+          inputsState.selectedDataInput.data.path.value.split('/')[0]
+      }).then(artifacts => {
+        const artifactsList = artifacts
+          .map(artifact => ({
+            label: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? '',
+            id: artifact.link_iteration
+              ? artifact.link_iteration.db_key
+              : artifact.key ?? ''
+          }))
+          .filter(artifact => artifact.label !== '')
+
+        inputsDispatch({
+          type: inputsActions.SET_ARTIFACTS,
+          payload: artifactsList
+        })
+      })
     }
   }, [
     fetchArtifacts,
     inputsState.artifacts.length,
     inputsState.newInput.path.project,
-    inputsState.newInputProjectPathEntered
+    inputsState.inputProjectPathEntered,
+    inputsState.selectedDataInput.data.path.value
   ])
 
   useEffect(() => {
-    if (pathScheme === MLRUN_STORAGE_INPUT_PATH_SCHEME) {
+    if (
+      inputsState.newInput.path.pathType === MLRUN_STORAGE_INPUT_PATH_SCHEME ||
+      inputsState.selectedDataInput.data.path.pathType ===
+        MLRUN_STORAGE_INPUT_PATH_SCHEME
+    ) {
       let matches = []
```

Comment:
remove this code to separate generateComboboxMatchesList function in jobsPanelDataInputs.utils